src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringScriptEngine.java

```diff
@@ -0,0 +1,40 @@
+package com.amazon.opendistroforelasticsearch.knn.plugin.script;
+
+import org.elasticsearch.script.ScoreScript;
+import org.elasticsearch.script.ScriptContext;
+import org.elasticsearch.script.ScriptEngine;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * KNN Custom scoring Engine implementation.
+ */
+public class KNNScoringScriptEngine implements ScriptEngine {
+
+    public static final String NAME = "knn";
+    public static final String SCRIPT_SOURCE = "knn_score";
+
+    @Override
+    public String getType() {
+        return NAME;
+    }
+
+    @Override
+    public <FactoryType> FactoryType compile(String name, String code, ScriptContext<FactoryType> context, Map<String, String> params) {
+        if (ScoreScript.CONTEXT.equals(context) == false) {
```

Comment:
How about simplify as `if (!ScoreScript.CONTEXT.equals(context))` ?

Is it possible that KNN script will have invalid context ? I see we have validate the script source in line 29, we need line 25 as the script context is possibly wrong?